src/implementations/truncation.jl

```diff
@@ -243,19 +70,49 @@ function findtruncated(values::AbstractVector, strategy::TruncationIntersection)
     inds = map(Base.Fix1(findtruncated, values), strategy.components)
     return intersect(inds...)
 end
+function findtruncated_sorted(values::AbstractVector, strategy::TruncationIntersection)
+    inds = map(Base.Fix1(findtruncated_sorted, values), strategy.components)
+    return intersect(inds...)
+end
+
+function findtruncated(values::AbstractVector, strategy::TruncationError)
+    I = sortperm(values; by=abs, rev=true)
+
+    Nᵖ = sum(Base.Fix2(^, strategy.p) ∘ abs, values)
+    ϵᵖ = max(strategy.atol^strategy.p, strategy.rtol^strategy.p * Nᵖ)
+    ϵᵖ ≥ Nᵖ && return I[Base.OneTo(0)]
+
+    truncerrᵖ = zero(real(eltype(values)))
+    rank = length(values)
+    for i in reverse(I)
+        @inbounds truncerrᵖ += abs(values[i])^strategy.p
+        if truncerrᵖ ≥ ϵᵖ
+            rank = i
+            break
+        end
+    end
+    return I[Base.OneTo(rank)]
+end
```

Comment:
Could this be written as:
```julia
function findtruncated(values::AbstractVector, strategy::TruncationError)
  I = sortperm(values; by=abs, rev=true)
  values_sorted = @view values[I]
  I_sorted = findtruncated_sorted(values_sorted, strategy)
  return I[I_sorted]
end
```
Just to avoid code duplication (I may have gotten the logic wrong but hopefully you get the idea).